libdino/src/service/connection_manager.vala

```diff
@@ -382,9 +382,11 @@ public class ConnectionManager : Object {
     }
 
     public static bool on_invalid_certificate(string domain, TlsCertificate peer_cert, TlsCertificateFlags errors) {
-        if (domain.has_suffix(".onion") && errors == TlsCertificateFlags.UNKNOWN_CA) {
+        if ((domain.has_suffix(".onion") || domain.has_suffix(".i2p") || domain.has_suffix(".loki")) && errors == TlsCertificateFlags.UNKNOWN_CA) {
```

Comment:
as @Quix0r pointed, you could something like
```vala
string [] domain_exceptions = {".onion", ".i2p", ".loki", ".etc"} 

foreach (var domain_exception in domain_exceptions) {
    if ( domain.has_suffix(domain_exception) && errors == TlsCertificateFlags.UNKNOWN_CA) {
      //do stuff
      break
    }
}
```

Reply:
I'll do that shortly.

Reply:
@PoneyClairDeLune Nice, and I wanted to add too that the `domain_exceptions` variable should have a more apt name like `tld_exceptions`, I think it's more readable

Reply:
@eerielili Done in https://github.com/dino/dino/pull/1445/commits/e4da09f90512e20c87c848b64eaa5844df7cec4a

Reply:
@eerielili Nice. Yeah, you should place inside the function, it looks cleaner and it'll work the same since they'll be in the same scope.
I'm not too big on OOP but I suggest declaring it as `private const string[]` at the top of the class _if_ needed in other functions, then you could call it with `this.tld_exceptions`.

Reply:
late to the party, but, can you add the `.local` to the list?

Reply:
@sergiuprt Just did it in [`859ead5d`](https://github.com/dino/dino/pull/1445/commits/859ead5db77cb019161bf94f198eb8cfb31cf6d9), but at this rate I doubt the Dino team would merge it in the foreseeable future. I've long been away from the XMPP ecosystem.